test/Util.java

```diff
@@ -199,4 +204,44 @@ public static ProcessExecutor createProcessExecutor(Path directory) {
                 .readOutput(true)
                 .destroyOnExit();
     }
+
+    @CommandLine.Command(name = "java")
+    private static class CLIOptions {
+        @CommandLine.Parameters String mainClass;
+        @CommandLine.Option(
+                names = {"--server"},
+                description = "Location of the archive containing a server artifact."
+        )
+        private String serverArchive;
+
+        @CommandLine.Option(
+                names = {"--console"},
+                description = "Location of the archive containing a console artifact."
+        )
+        private String consoleArchive;
+
+        public String getServerArchive() {
+            return serverArchive;
+        }
+
+        public String getConsoleArchive() {
+            return consoleArchive;
+        }
+
+        public static Optional<CLIOptions> parseCLIOptions(String[] args) {
+            CommandLine commandLine = new CommandLine(new CLIOptions());
+            try {
+                CommandLine.ParseResult result = commandLine.parseArgs(args);
+                return Optional.of(result.asCommandLineList().get(0).getCommand());
```

Comment:
If the parse result succeeded, we get the first command in the commandLineList and return it in an optional.